MSVC: Improve compiler flags

SmallJoker opened the pull request:
- Use equivalent /Ox for releases
- Make Release builds generate debug information

kwolekr commented:
:-1:

Debug needs to be unoptimized or else it becomes difficult if not impossible to follow the code in a debugger.
Semidebug attempts to be something in between Debug and Release with -O1 that trades some debugability for speed, which is necessary in some cases.

Additionally, there's an entirely unrelated code change in this PR.

SmallJoker commented:
The problem is that the current debug build is simply not playable due, barely 1 FPS (maybe more when the ABMs finished after 12 seconds).
However, I reverted the changes in the *Debug parameters, so it only will generate debug information for releases now. [Stackoverflow reference](http://stackoverflow.com/questions/28178978/how-to-generate-pdb-files-for-release-build-with-cmake-flags#31264946)

The two changed lines in aa70237 came from a fix while debugging and somehow found its way in here. Removed now :)

kwolekr commented:
And debug builds with optimizations are rarely helpful in, you know, debugging.  Like I said, if you have a problem that requires to use the debugger in-game, there's the Semidebug build.

FWIW, I always run MT with full debug and it works fine for me.  Here's a nickel, kid.  Buy yourself a better computer.

kwolekr commented on src/CMakeLists.txt:
```diff
@@ -708,7 +708,7 @@ if(MSVC)
 	# Flags for C files (sqlite)
 	# /MT = Link statically with standard library stuff
-	set(CMAKE_C_FLAGS_RELEASE "/O2 /Ob2 /MT")
+	set(CMAKE_C_FLAGS_RELEASE "/MT /O2 /Ob2")
```
Does changing the order of these flags do anything?

SmallJoker replied:
No, it doesn't. I wanted to brin in some order but removed this change right now to simplify this pull.

kwolekr commented:
:+1:
